File: Easy/2025-12-24-3074-Apple-Redistribution-into-Boxes/solution.js
function minBoxes(apple, capacity) {
    const totalApples = apple.reduce((a, b) => a + b, 0);
    capacity.sort((a, b) => b - a);
    
    let boxesUsed = 0;
    let currentCapacity = 0;

    for (let cap of capacity) {
        if (currentCapacity >= totalApples) break;
        currentCapacity += cap;
        boxesUsed++;
    }

    return boxesUsed;
}

File: Easy/2025-12-24-3074-Apple-Redistribution-into-Boxes/solution.py
class Solution:
    def minBoxes(self, apple: List[int], capacity: List[int]) -> int:
        total_apples = sum(apple)
        capacity.sort(reverse=True)
        
        boxes_used = 0
        current_capacity = 0
        
        for cap in capacity:
            current_capacity += cap
            boxes_used += 1
            if current_capacity >= total_apples:
                return boxes_used
        
        return boxes_used